shared/subspace-kzg/src/lib.rs

```diff
@@ -364,11 +682,18 @@ pub struct Kzg {
 }
 
 impl Kzg {
-    /// Create new instance with given KZG settings.
+    /// Create new instance with embedded KZG settings.
     ///
-    /// Canonical KZG settings can be obtained using `embedded_kzg_settings()` function that becomes
-    /// available with `embedded-kzg-settings` feature (enabled by default).
-    pub fn new(kzg_settings: FsKZGSettings) -> Self {
+    /// NOTE: Prefer cloning to instantiation since cloning is cheap and instantiation is not!
+    #[allow(
+        clippy::new_without_default,
+        reason = "Caller really should read the function description"
+    )]
```

Comment:
We're now using a compiler version which supports [`expect(lint)`](https://doc.rust-lang.org/reference/attributes/diagnostics.html#lint-check-attributes). If we use expect, and someone implements default, the absence of this lint will cause a compilation failure. Which seems to be what we want here.

Nit: the reasoning behind using `expect` rather than `allow` is not immediately clear from the function description. Being explicit will make this easier to catch during a review or after a compilation failure.

```suggestion
    #[expect(
        clippy::new_without_default,
        reason = "Default must not be implemented, because Kzg should be cloned instead. Cloning is cheap and instantiation is not."
    )]
```